test/test_interpolate_project.py

```diff
@@ -2,25 +2,38 @@
 from mpi4py import MPI
 import numpy as np
 import pyleopart
+import pytest
 
-
-def test_interpolate_project():
+# Test back-and-forth projection on 2D mesh
+@pytest.mark.parametrize("is_vec, k", [(True, 2), (True, 3), (False, 2), (False, 3)])
```

Comment:
I think it would be better to just have two tests, one for scalar and one for vector, rather than a load of `if` statements in the test.

Reply:
Now split into scalar and vector test.